handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -760,16 +769,13 @@ public final SSLEngineResult unwrap(
         }
 
         synchronized (this) {
-            // Check to make sure the engine has not been closed
-            if (isDestroyed()) {
+            // Check to make sure the engine has not been closed and the inbound was not marked as done.
+            if (isDestroyed() || (handshakeState == HandshakeState.NOT_STARTED && isInboundDone())) {
```

Comment:
nit: you have parens here but not on line 590 ... can we be consistent? strictly speaking they aren't necessary but as long as we are consistent its all good.